api/v1/checks.go

```diff
@@ -163,6 +163,37 @@ func (c S3BucketCheck) GetType() string {
 	return "s3Bucket"
 }
 
+type ResticCheck struct {
+	// Description about the current restic canary check
+	Description string `yaml:"description,omitempty" json:"description,omitempty"`
+	// Repository The restic repository path eg: rest:https://user:pass@host:8000/ or rest:https://host:8000/ or s3:s3.amazonaws.com/bucket_name
+	Repository string `yaml:"repository" json:"repository"`
+	// Password for the restic repository
+	Password string `yaml:"password" json:"password"`
+	// MaxAge for backup freshness
+	MaxAge string `yaml:"maxAge" json:"maxAge"`
+	// CheckIntegrity when enabled will check the Integrity and consistency of the restic reposiotry
```

Comment:
Is this repository or the latest snapshot ?

Reply:
The MaxAge is for the latest snapshot.... the integrity check is for the whole repository

Reply:
We don't want to run an integrity check on the whole repo, as it can easily grow to TB's - Only on the most recent snapshot